### ISSUE TITLE
Add worker beacon

### ISSUE THREAD

**Issue by mingyc** [WICG/unload-beacon#39]
https://wicg.github.io/unload-beacon/#issue-fdb03e89

> ISSUE 2:  Add worker beacons as well?

It doesn't make sense to support all types of workers but only dedicated worker, as a beacon has to be associated with a browsing context and it must not be shared across different context for privacy reason.

**Comment by Sora2455** [comment]
In that case, wouldn't the "browsing context" of the dedicated worker be the last browsing context that had the dedicated worker open?

**Comment by mingyc** [comment]
Yes. What I mean is to support only dedicated worker. Updated to clarify it.

**Comment by mingyc** [comment]
Correction to my [comment](https://github.com/WICG/pending-beacon/issues/39#issuecomment-1260564170) above. There are some concerns around the lifetime of a PendingBeacon `P` created in a dedicated worker `W`. Trying to list possible options below:

1. The same as `W`'s lifetime, meaning that `P` should be sent out or discarded when `W` is closed.
   - Easiest to understand and use.
   - But in some cases `W` can live longer than its parent browsing context (Document, or another DedicatedWorker), which might lead to prviacy issues.
2. The same as the ancestor Document `D` (which owns `W` directly or inderectly via nested dedicated workers)'s lifetime.
   - This means every PendingBeacon `P` created in `W` should be attached to `D`.
   - If `W` is closed earlier before `D` is destroyed, it's ok.
   - If `W` stays active after `D` is destroyed, anything happens to `P` must be no-op, and no new PendingBeacon can be created.
     - There might be some undesired contentions happen?
3. The same as parent browsing context (either a Document or a dedicated worker):
   - If parent is dedicated worker, it does not make sense to attach a PendingBeacon to it.


To me it looks like Option 2 is the one closest to the current PendingBeacon behavior.

**Comment by fergald** [comment]
If W can still make network requests then it should be able to create a new Beacon and have it send when it is finally destroyed.

**Comment by mingyc** [comment]
Ok so basically Option 1. In such case, `backgroundTimeout` should be disallowed as there is no page visibility event.

**Comment by fergald** [comment]
I'm not sure about that disallowing `backgroundTimeout`. It can still refer to the ancestor frame. We shouldn't make it hard to push code into a worker unless there's a reason why implementing it would be a lot of work.
